README.md

```diff
@@ -1,38 +1,96 @@
 # py-es-bulk
-A simple wrapper around elasticsearch-py client streaming_bulk() API with
-robust error handling.
+A simple wrapper around the Python `elasticsearch` client `put_template()`, `streaming_bulk()`, and `parallel_bulk()` helper APIs with robust error handling.
 
 This library is designed to work across various versions of the
-elasticsearch Python module and of the Elasticsearch server, as well as
-Elasticsearch V1 with the elasticsearch1 Python module.
+elasticsearch Python module and of the Elasticsearch server, by
+dynamically identifying the module used to create the `Elasticsearch` object.
 
 These names are available for import:
 
-* put_template
+* `put_template`
 
     Push a document template to the server using a specified
     Elasticsearch object. This module will determine whether
     a template document of the same name and version already
     exists, and PUT the new template if not.
 
-* streaming_bulk
+    Args:
+    - `es`: An instance of the `Elasticsearch` class.
+    - `name`: The name of the template.
+    - `mapping_name`: The name of the mapping used in the template.
+    - `body`: The payload body of the template.
+
+    Returns: A tuple (start_time, end_time, retry_count, error_keys)
+
+* `streaming_bulk`
 
     Push multiple source documents to Elasticsearch indices,
     using proper error handling and retry logic.
 
-* parallel_bulk
+    Args:
+    - 'es': An instance of the `Elasticsearch` class.
+    - `actions`: An iterable of Elasticsearch action records (passed directly to Elasticsearch).
+    - `errorsfp`: A file pointer where HTTP 400 errors are logged.
+    - `logger`: A `Logger` object where messages can be logged.
+
+    Returns: A tuple (start_time, end_time, successfully_indexed, duplicate, failed, retry_count).
+
+
+* `parallel_bulk`
 
     Push multiple source documents to Elasticsearch indices
     in parallel across multiple threads, using proper error
     handling and retry logic.
 
-* TemplateException
+    Args:
+    - `es`: An instance of the `Elasticsearch` class.
+    - `actions`: An iterable of Elasticsearch action records
+    (passed directly to Elasticsearch)
+    - `errorsfp`: A file pointer where HTTP 400 errors are logged.
+    - `logger`: A `Logger` object where messages can be logged.
+    - `chunk_size=10000000`: Number of docs sent in one chunk to Elasticsearch.
+    - `max_chunk_bytes=104857600`: The maximum size of of a request.
```

Comment:
`s/of of/of/`

Reply:
Ugh. Yeah; plus, the unit tests didn't work because the module wasn't found. I have no idea what happened there, but I might have broken the packaging a different way that slipped past on my local testing.